### ISSUE TITLE
RTL: floated elements not visible in PDF

### ISSUE THREAD

**Issue by sagarvora** [Kozea/WeasyPrint#1110]
Issue:
Elements with the `float` CSS property do not render properly for RTL. I think it has to do with some of the changes introduced in #1079. I think weasyprint internally changes `float: left` to mean `float: right` for RTL but I don't think that's expected behavior.

Weasyprint `v51` Output:

![Screenshot from 2020-05-12 08-15-34](https://user-images.githubusercontent.com/16315650/81632838-dae0f500-9428-11ea-8b64-90a53d072df6.png)

Weasyprint `master` branch Output:

![Screenshot from 2020-05-12 08-09-18](https://user-images.githubusercontent.com/16315650/81632885-f5b36980-9428-11ea-9096-a00497bd57bd.png)

Expected Output (rendered using `wkhtmltopdf`):

![Screenshot from 2020-05-12 08-02-12](https://user-images.githubusercontent.com/16315650/81632913-08c63980-9429-11ea-9324-015d58494f1e.png)

Source Code:
https://gist.github.com/sagarvora/b3841a2852971ed4ceaf4eb7564b588b

**Comment by liZe** [comment]
Hello!

Your source code doesn’t correspond to the images you’ve included. Could you please attach the right code, with floating elements?

**Comment by liZe** [comment]
@sagarvora I’ve tried to add the Boostrap stylesheet, but it does’t work. I think that a stylesheet is missing.

**Comment by liZe** [comment]
Here’s a sample that used to work, but doesn’t anymore. I assume that it’s the same bug as yours:
```html
<div style="direction: rtl">
  <div style="float: right">abc</div>
  <div style="float: left">def</div>
</div>
```

**Comment by liZe** [comment]
#1079 helped to handle more rtl cases, but it broke some of them, including floats in rtl containers. f6f9f14 should fix all cases given in the spec for boxes avoiding floats (other floats, line boxes, tables, etc) and add tests to avoid these issues to come back.

@sagarvora I’d be interested in real-life tests, let me know if it fixes your documents!

**Comment by sagarvora** [comment]
@liZe Yes, we do have expected output! Tested on production as well. Sorry for a bad example (I removed a stylesheet, should've mentioned that). Thanks for all the help :heart: 

![Screenshot from 2020-05-23 02-07-28](https://user-images.githubusercontent.com/16315650/82707685-2ce01100-9c9a-11ea-9658-0231a8eb965f.png)

**Comment by sagarvora** [comment]
![Screenshot from 2020-05-23 02-07-28](https://user-images.githubusercontent.com/16315650/82708018-eb9c3100-9c9a-11ea-9722-617fc18de69a.png)

@liZe This is a minor unrelated issue which can be easily fixed by setting a custom width. But it works for English.

English:
![Screenshot from 2020-05-23 02-21-15](https://user-images.githubusercontent.com/16315650/82708602-4b470c00-9c9c-11ea-9092-e953acd8feaf.png)

Arabic:
![Screenshot from 2020-05-23 02-20-20](https://user-images.githubusercontent.com/16315650/82708643-5f8b0900-9c9c-11ea-8819-b33ff4b06df7.png)


Basically, the table columns don't resize to fit content in Arabic like they do in English.

**Comment by liZe** [comment]
> This is a minor unrelated issue which can be easily fixed by setting a custom width. But it works for English. Basically, the table columns don't resize to fit content in Arabic like they do in English.

Could you please provide the HTML+CSS sample showing this problem?
